Vagrantfile

```diff
@@ -1,11 +1,23 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
+class Config
+	def readfunction
+		ne = IO.readlines('settings.sh')		
+		nm = ne[1].split "="		
+		na = nm[1].split "/"
+		nip = na[0][0..-2]
+		$ip = nip + 18.to_s
+	end
+
+obj = Config.new()
+obj.readfunction
+
```

Comment:
Why do you make things so complicated? You can use `load settings.sh`, which includes that file in the ruby script (similar to `source settings.sh` which includes it in a bash script). I don't know ruby well, but I searched and I found this solution.

Then, in `settings.sh`, you can use `NETWORK="192.168.10"`, instead of `NETWORK=192.168.10`, so that it is valid both in ruby and in bash. Also be aware that you should not use spaces around `=`, otherwise it will not be valid in bash.